### DISCUSSION TITLE
Upsert not working

### DISCUSSION THREAD

**Discussion by krishnaram** [MorphiaOrg/morphia#2407] [General]
We are using 

```xml
<groupId>dev.morphia.morphia</groupId>
<artifactId>morphia-core</artifactId>
<version>2.3.1</version>
```

```java
UpdateResult result = morphiaDataStore.find(DataVolumeConfiguration.class)
  .filter((Filters.eq(HelperUtils.ID_KEY,updateKey)))
  .update(UpdateOperators.set(resetDates, fleetConfiguration.getResetDates()),
          UpdateOperators.set(limitOverall, fleetConfiguration.getLimitOverall()),
          UpdateOperators.set(limitPredictionStartsFrom, fleetConfiguration.getLimitPredictionStartsFrom()),
          UpdateOperators.set(intervalUpdateCache, fleetConfiguration.getIntervalUpdateCache()))
  .execute(new UpdateOptions().upsert(true));
	
log.info("{} record got upserted, {}: ID is ",result.getMatchedCount(), result.getUpsertedId());
```

Logs we are getting
`0 record got upserted, BsonString{value=eVehicle''}: ID is `

Our "_id" is not autogenerated.

Please anyone can help on this issue. Thanks

**Comment by evanchooly** [comment]
it looks like perhaps that log entry didn't get completely copied in to the comment here.  Everything looks right (though your filter has extra parentheses which shouldn't matter).  If you could put together a [reproducer](/morphiaorg/reproducer) I'd be happy to take a look and see.

**Reply by krishnaram** [reply]
This is logs I am getting for result 
`log.info("{} record got upserted, {}: ID is ",result.getMatchedCount(), result.getUpsertedId());`

There is no exception or other logs after upsert. Upsert is not happening.

**Reply by krishnaram** [reply]
We have @Id our own  key,  @Version in morphia entity. Does it effecting in creating record?

**Comment by krishnaram** [comment]
I found the issue. Upsert only work when "_id" is autogenerated. If it is given by the entity it won't create records in upsert.
